axiom/tests/test_log_events.c

```diff
@@ -142,33 +161,125 @@ static void test_events_sample(void) {
+  nr_log_event_set_timestamp(e, (LOG_TIMESTAMP + 100) * NR_TIME_DIVISOR_MS);
+  event_dropped = nr_log_events_add_event(events, e);
+  tlib_pass_if_true("final event caused drop", event_dropped,
+                    "nr_log_events_add_event: got [%d], want [%d]",
+                    event_dropped, true);
+
+  // verify two highest priority events were retained
+  // convert log event pool to a vector and test
+  vector = nr_vector_create(10, NULL, NULL);
+  nr_log_events_to_vector(events, vector);
+  pass = nr_vector_get_element(vector, 0, &test_e);
+  tlib_pass_if_true("retrived log element from vector OK", pass,
+                    "expected TRUE");
+  tlib_pass_if_str_equal("LOG_MESSAGE_4 log event retained", LOG_MESSAGE_4,
+                         ((nr_log_event_t*)test_e)->message);
+  tlib_pass_if_int_equal("Priority 50 log event retained", 50,
+                         ((nr_log_event_t*)test_e)->priority);
+  pass = nr_vector_get_element(vector, 1, &test_e);
+  tlib_pass_if_true("retrived log element from vector OK", pass,
+                    "expected TRUE");
+  tlib_pass_if_str_equal("LOG_MESSAGE_0 log event retained", LOG_MESSAGE_0,
+                         ((nr_log_event_t*)test_e)->message);
+  tlib_pass_if_int_equal("Priority 100 log event retained", 100,
+                         ((nr_log_event_t*)test_e)->priority);
+
+  nr_vector_destroy(&vector);
+
   nr_log_events_destroy(&events);
   tlib_pass_if_null("events destroyed", events);
 }
 
+static void test_log_event_comparator(void) {
+  nr_log_event_t log0 = {.message = "LOG0", .priority = 30, .timestamp = 0};
+  nr_log_event_t log1 = {.message = "LOG1", .priority = 20, .timestamp = 0};
+  nr_log_event_t log2 = {.message = "LOG2", .priority = 10, .timestamp = 0};
+  nr_log_event_t log3 = {.message = "LOG3", .priority = 0, .timestamp = 0};
+  nr_log_event_t log4 = {.message = "LOG4", .priority = 0, .timestamp = 100};
+
+  nr_vector_t* events = nr_vector_create(12, NULL, NULL);
+
+  /*
+   * The comparator function is tested by using it to sort a vector of
+   * log events. In this way, all necessary test cases are covered.
+   *
+   * The comparator first compares a log events's priority, which is a bit
+   * field with bits set according to NR_SEGEMENT_PRIORITY_* flags. The
+   * priority with the higher numerical value is considered higher.
+   *
+   * If the priorities of two events are the same, the comparator
+   * compares the events age duration. The younger event is considered
+   * higher.
+   *
+   * The table below shows the final ordering and the respective values
+   * that are considered by the comparator function.
+   *
+   * Position | Message    | Priority  | Timestamp
+   * ---------+------------+-----------+----------
+   * 4        | "LOG0"     | 30        | 0
+   * 3        | "LOG1"     | 20        | 0
+   * 2        | "LOG2"     | 10        | 0
+   * 1        | "LOG4"     | 0         | 100
+   * 0        | "LOG3"     | 0         | 0
+   */
```

Comment:
```suggestion
   */
   // clang-format on
```

Reply:
I don't think this will be reformatted I kept the line lengths short enough.